spring-cloud-task-batch/src/test/java/org/springframework/cloud/task/batch/handler/TaskJobLauncherCommandLineRunnerTests.java

```diff
@@ -0,0 +1,240 @@
+							TaskJobLauncherAutoConfiguration.class }, enabledArgs);
+		}
+		catch (IllegalStateException exception) {
+			isExceptionThrown = true;
+		}
+		assertThat(isExceptionThrown).isFalse();
+		validateContext();
+	}
+
+	@Test
+	public void testCommandLineRunnerSetToFalse() {
+		String[] enabledArgs = new String[] { "--spring.cloud.task.batch.command.line.runner.enabled=false",
+				"--spring.cloud.task.closecontext_enable=false" };
+		this.applicationContext = SpringApplication
+				.run(new Class[] { TaskJobLauncherCommandLineRunnerTests.JobConfiguration.class,
+						PropertyPlaceholderAutoConfiguration.class,
+						EmbeddedDataSourceConfiguration.class,
+						BatchAutoConfiguration.class,
+						TaskBatchAutoConfiguration.class,
+						TaskJobLauncherAutoConfiguration.class }, enabledArgs);
+		validateContext();
+		assertThat(applicationContext.getBean(JobLauncherCommandLineRunner.class)).isNotNull();
+		boolean exceptionThrown = false;
+		try {
+			applicationContext.getBean(TaskJobLauncherCommandLineRunner.class);
+		}
+		catch (NoSuchBeanDefinitionException exception) {
+			exceptionThrown = true;
+		}
+		assertThat(exceptionThrown).isTrue();
+		validateContext();
+	}
+
+	private void validateContext() {
+		TaskExplorer taskExplorer = this.applicationContext.getBean(TaskExplorer.class);
+
+		Page<TaskExecution> page = taskExplorer.findTaskExecutionsByName("application", PageRequest.of(0, 1));
+
+		Set<Long> jobExecutionIds = taskExplorer
+				.getJobExecutionIdsByTaskExecutionId(page.iterator().next().getExecutionId());
+
+		assertThat(jobExecutionIds.size()).isEqualTo(1);
+		assertThat(taskExplorer.getTaskExecution(jobExecutionIds.iterator().next()).getExecutionId()).isEqualTo(1);
+
+	}
+
+	@Configuration
+	@EnableBatchProcessing
+	@EnableTask
+	public static class JobConfiguration {
+
+		@Autowired
+		private JobBuilderFactory jobBuilderFactory;
+
+		@Autowired
+		private StepBuilderFactory stepBuilderFactory;
+
+		@Bean
+		public Job job() {
+			return jobBuilderFactory.get("job")
+					.start(stepBuilderFactory.get("step1").tasklet(new Tasklet() {
+						@Override
+						public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext)
+								throws Exception {
+							System.out.println("Executed");
+							return RepeatStatus.FINISHED;
+						}
+					}).build())
+					.build();
+		}
+	}
+
+	@Configuration
+	@EnableBatchProcessing
+	@EnableTask
+	public static class JobWithFailureConfiguration {
+
+		@Autowired
+		private JobBuilderFactory jobBuilderFactory;
+
+		@Autowired
+		private StepBuilderFactory stepBuilderFactory;
+
+		@Bean
+		public Job jobFail() {
+			return jobBuilderFactory.get("jobA")
+					.start(stepBuilderFactory.get("step1").tasklet(new Tasklet() {
+						@Override
+						public RepeatStatus execute(StepContribution contribution,
+								ChunkContext chunkContext)
+								throws Exception {
+							System.out.println("Executed");
+							if (true) {
```

Comment:
Why the if?

Reply:
Compiler states that we have unreachable code if the exception is thrown with out some kind of condition prior to the return.

Reply:
Yeah...because you still have the return.  I believe just throwing the exception (no return) is valid...